perl/meson_options.txt

```diff
@@ -0,0 +1,50 @@
+# Nix-Perl build options
+#============================================================================
+
+
+# compiler args
+#============================================================================
+
+option(
+    'version',
+    type : 'string',
+    description : 'nix-perl version')
+
+option(
+    'tests',
+    type : 'feature',
+    value : 'disabled',
+    description : 'run nix-perl tests')
+
+option(
+    'errorflags',
+    type : 'array',
+    value : [
+        '-Wno-pedantic',
+        '-Wno-non-virtual-dtor',
+        '-Wno-unused-parameter',
+        '-Wno-variadic-macros',
+        '-Wdeprecated-declarations',
+        '-Wno-missing-field-initializers',
+        '-Wno-unknown-warning-option',
+        '-Wno-unused-variable',
+        '-Wno-literal-suffix',
+        '-Wno-reserved-user-defined-literal',
+        '-Wno-duplicate-decl-specifier',
+        '-Wno-pointer-bool-conversion'],
+    description : 'C build flags')
```

Comment:
This is new to this build system, right? If so, I think we should probably skip for now and add later, since it is a "new feature".

Reply:
it isnt new, its already in the make files. this is like for like as close as possible

Reply:
They're not behind an option though, they're just universally turned on. If the maintainers like this apporach that's fine, but it sure feels llike the `add_project_arguments()` approach I suggested initially would be a more 1:1 mapping of what the Makefiles had

Reply:
switched to the static setup that is already in the build for now. maybe in the future this could cause issues if any of those flags needed overriding, but matching the current build is probably the best idea for porting;- improvements can build from that